spec/cost_calculator_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "cost_calculator"
+
+RSpec.describe "CostCalculator" do
+
+  before(:each) do
+    @calc = CostCalculator.new
+  end
+
+  describe "#price_for_dish" do
+    it "gets the price for an ordered dish from the menu" do
+      @menu = double("menu", menu_contents: { "Fish" => 2 })
+      @order = double("order", current_order: { "dish" => "Fish", "quantity" => 3 })
+      expect(@calc.price_for_dish(@order.current_order, @menu.menu_contents)).to eq(2)
```

Comment:
You don't actually need doubles here as you are passing hashes, and not objects into your function. You could write:
```
expect(@calc.price_for_dish({ "dish" => "Fish", "quantity" => 3 }, { "Fish" => 2 })).to eq(2)
```

**However** this feels like a bit of a design flaw. You probably should pass objects to that function directly, and the function would then call the necessary methods on them. And in that case, you would indeed need doubles.